### PR TITLE
Improvements for App support: add authP modifiers in AragonApp base and add more syntax sugar

### PR DESCRIPTION
**Pull request by izqui**
Looks like we might need to release aOS 3.0.1 earlier than expected 😅

### CODE REVIEW

File: contracts/apps/AragonApp.sol [hunk 1]
```diff
@@ -3,15 +3,29 @@ pragma solidity ^0.4.18;
...
 
-    function canPerform(address _sender, bytes32 _role) public view returns (bool) {
-        return address(kernel) == 0 || kernel.hasPermission(_sender, address(this), _role, new bytes(0));
+    modifier authP(bytes32 _role, uint256[] params) {
+        require(canPerform(msg.sender, _role, params));
+        _;
+    }
+
+    function canPerform(address _sender, bytes32 _role, uint256[] params) public view returns (bool) {
+        bytes memory how; // no need to init memory as it is never used
```
Comment (sohkai) [inline contracts/apps/AragonApp.sol]: We now have two slightly different ways of doing this `uint[]` to `bytes` conversion (see `canPerform()` in `Kernel`), so I'm wondering if it'd be nice to add this as `toBytes()` somewhere. Maybe we can move some of the more general helpers in `ScriptHelpers` (e.g. `uint256At`, `addressAt`, `uint32At`, `locationOf`) to another library like `BytesHelpers`?

File: contracts/apps/AragonApp.sol [hunk 2]
```diff
@@ -3,15 +3,29 @@ pragma solidity ^0.4.18;
...
+        if (params.length > 0) {
+            uint256 byteLength = params.length * 32;
+            assembly {
+                how := params // forced casting
+                mstore(how, byteLength)
+            }
+        }
+        return address(kernel) == 0 || kernel.hasPermission(_sender, address(this), _role, how);
```
Comment (sohkai) [inline contracts/apps/AragonApp.sol]: This should probably be `address(kernel) != 0 && ...`?

`getCode()` will fail if `address(kernel) == 0` so you won't be able to perform the action anyway.

Comment (izqui) [reply contracts/apps/AragonApp.sol]: this is for the case when you are using the app without a proxy

File: contracts/apps/AragonApp.sol (hunk 1 above)
Comment (izqui) [reply contracts/apps/AragonApp.sol]: Good idea, making this its own issue

**Pull request merged**